### PR TITLE
Added real-time handling for control value changes to prevent the nee…

### PR DESCRIPTION
**Pull request by udidol**
…d for re-rendering onElementChange - Image Carousel widget

### CODE REVIEW

File: assets/dev/js/frontend/handlers/image-carousel.js [hunk 1]
```diff
@@ -121,14 +122,104 @@ class ImageCarouselHandler extends elementorModules.frontend.handlers.Base {
...
-			this.swiper.update();
+		const changeableProperties = this.getChangeableProperties();
+		// const currentDeviceMode = elementorFrontend.getCurrentDeviceMode();
+
+		// Handle responsive controls
+		/*if ( propertyName.includes( '_' + currentDeviceMode ) ) {
+			this.updateResponsiveSwiperOption( propertyName, currentDeviceMode );
+		}*/
+
+		if ( changeableProperties.hasOwnProperty( propertyName ) ) {
```
Comment (kobizz) [inline assets/dev/js/frontend/handlers/image-carousel.js:221]:
```suggestion
		if ( changeableProperties.propertyName ) {
```

File: assets/dev/js/frontend/handlers/image-carousel.js [hunk 2]
```diff
@@ -121,14 +122,104 @@ class ImageCarouselHandler extends elementorModules.frontend.handlers.Base {
...
+
+		// 'pause_on_hover' is implemented by the handler with event listeners, not the Swiper library
+		if ( 'pause_on_hover' !== propertyName ) {
+			this.swiper.params[ propertyToUpdate ] = valueToUpdate;
+		}
+
+		this.swiper.update();
+	}
+
+	/*updateResponsiveSwiperOption( propertyName, currentDeviceMode ) {
```
Comment (kobizz) [inline assets/dev/js/frontend/handlers/image-carousel.js:177]: What about responsive handling?

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/handlers/image-carousel.js (hunk 2 above)
Comment (udidol) [reply assets/dev/js/frontend/handlers/image-carousel.js:177]: It doesn't work. `swiper.update()` doesn't update responsive values, I think it's a Swiper bug, I opened an issue in the Swiper github: https://github.com/nolimits4web/swiper/issues/3539

Comment (udidol) [reply assets/dev/js/frontend/handlers/image-carousel.js:177]: I left all responsive controls to re-render from PHP for now

Comment (kobizz) [reply assets/dev/js/frontend/handlers/image-carousel.js:177]: Then remove all the breakpoints related and commented-out code. You can keep it in a stash for future use

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/handlers/base-swiper.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import BaseHandler from './base';
```
Comment (kobizz) [inline assets/dev/js/frontend/handlers/base-swiper.js:1]: If there is common settings/elements, consider moving them here.

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/handlers/base-swiper.js (hunk 3 above)
Comment (udidol) [reply assets/dev/js/frontend/handlers/base-swiper.js:1]: There aren't any. If we make a decision to change control names to unify all of the Swiper widgets, we would probably be able to move a lot more to the base. Unfortunately, this is currently not possible

**Review by KingYes — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**